astroquery/alma/core.py

```diff
@@ -280,33 +287,50 @@ def stage_data(self, uids):
+                                     "the ALMA servers.")
                 else:
                     raise
-            if jdata['type'] != 'PROJECT':
-                log.error("Skipped uid {uu} because it is not a project and"
-                          "lacks the appropriate metadata; it is a "
-                          "{jdata}".format(uu=uu, jdata=jdata['type']))
-                continue
-            table = uid_json_to_table(jdata)
-            table['sizeInBytes'].unit = u.B
-            table.rename_column('sizeInBytes', 'size')
-            table.add_column(Column(data=['{dataarchive_url}/dataPortal/{name}'
-                                          .format(dataarchive_url=dataarchive_url,
-                                                  name=name)
-                                          for name in table['name']],
-                                    name='URL'))
-
-            isp = self.is_proprietary(uid)
-            table.add_column(Column(data=[isp for row in table],
-                                    name='isProprietary'))
-
-            tables.append(table)
-            log.debug("Completed metadata retrieval for {0}".format(uu))
+
+            if return_json:
+                tables.append(jdata)
+            else:
+                if jdata['type'] != 'PROJECT':
+                    log.error("Skipped uid {uu} because it is not a project and"
+                              "lacks the appropriate metadata; it is a "
+                              "{jdata}".format(uu=uu, jdata=jdata['type']))
+                    continue
+                if expand_tarfiles:
+                    table = uid_json_to_table(jdata, productlist=['ASDM',
+                                                                  'PIPELINE_PRODUCT'])
+                else:
+                    table = uid_json_to_table(jdata,
+                                              productlist=['ASDM',
+                                                           'PIPELINE_PRODUCT'
+                                                           'PIPELINE_PRODUCT_TARFILE',
+                                                           'PIPELINE_AUXILIARY_TARFILE'])
+                table['sizeInBytes'].unit = u.B
+                table.rename_column('sizeInBytes', 'size')
+                table.add_column(Column(data=['{dataarchive_url}/dataPortal/{name}'
+                                              .format(dataarchive_url=dataarchive_url,
+                                                      name=name)
+                                              for name in table['name']],
+                                        name='URL'))
+
+                isp = self.is_proprietary(uid)
+                table.add_column(Column(data=[isp for row in table],
+                                        name='isProprietary'))
+
+                tables.append(table)
+                log.debug("Completed metadata retrieval for {0}".format(uu))
 
         if len(tables) == 0:
```

Comment:
Indent looks bad here, but I'm certain it's just a bug in the github app.

Reply:
yeah, I don't see a problem